### PR TITLE
docs: add pytest-otel details

### PR DESCRIPTION
**Pull request by kuisathaverat**
it adds pytest-otel details

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
Can we please backport to 7.17?

File: docs/en/observability/ci-cd-observability.asciidoc [hunk 1]
```diff
@@ -400,6 +400,29 @@ image::images/jenkins-makefile.png[Jenkins build executing an instrumented Makef
 .ServiceMap view of a Jenkins pipeline execution instrumented with the otel-cli
 image::images/jenkins-service-map.png[Jenkins service map view]
 
+[discrete]
+[[ci-cd-pytest-otel]]
+=== Pytest-otel
+
+https://pypi.org/project/pytest-otel/[pytest-otel] is a pytest plugin for sending Python test
+results as OpenTelemetry traces. The test traces help to understand tests execution, 
+detect bottlenecks, and compare test executions across time to detect misbehaviors and issues.
```
Comment (dedemorton) [inline docs/en/observability/ci-cd-observability.asciidoc:409]:
```suggestion
results as OpenTelemetry traces. The test traces help you understand test execution, 
detect bottlenecks, and compare test executions across time to detect misbehaviors and issues.
```

**Review by dedemorton — COMMENTED** [review]
Couple of minor editorial changes. Otherwise looks good!

**Comment by dedemorton** [comment]
@kuisathaverat Hope you don't mind, but I've added backport labels because I wanted to make sure the backport gets into 8.1 (we just created the branch a few min ago).

**Pull request merged**